File: src/main/java/com/jss/jssbackend/modules/tiers/service/AttachmentTypeService.java
package com.jss.jssbackend.modules.tiers.service;

import java.util.List;

import com.jss.jssbackend.modules.tiers.model.AttachmentType;

public interface AttachmentTypeService {
    public List<AttachmentType> getAttachmentTypes();

    public AttachmentType getAttachmentType(Integer id);
}

File: client/src/app/modules/tiers/services/attachment.type.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/appRest.service';
import { AttachmentType } from '../../tiers/model/AttachmentType';

@Injectable({
  providedIn: 'root'
})
export class AttachmentTypeService extends AppRestService<AttachmentType>{

  constructor(http: HttpClient) {
    super(http, "tiers");
  }

  getAttachmentTypes() {
    return this.getList(new HttpParams(), "attachment-types");
  }

}
